### DISCUSSION TITLE
Lack of build parallelisation

### DISCUSSION THREAD

**Discussion by OllyFunkster** [rocm-arch/rocm-arch#778] [Support] [answered]
Hi,

I notice that some packages (e.g. rocsparse) only use one thread when building. This seems like it wouldn't be intentional. What info would be required to figure out why this is?

**Comment by JP-Ellis** [comment, marked as answer]
The PKGBUILDs actually just follow the default set by your `makepkg.conf` and the upstream package.  If you want to enable parallelisation by default when building packages, you should set `MAKEFLAGS="-j$(nproc)"` inside your `makepkg.conf`.

**Comment by OllyFunkster** [comment]
Ahhh thank you! I feel daft for not knowing that, but I guess I was lulled into a false sense of security by the parallellisation done during building LLVM.

I appreciate you taking the time to put me on the right track  :)

**Reply by lahwaacz** [reply]
The default behaviour depends on the build system. For example, Ninja uses all cores by default, whereas GNU Make uses one core by default.
